File: server/datasets/python_source-code/Multiple-E4/type3_composite_1_substring_search.py
def get_substring_index(text_str, target_str):
    if not target_str: return 0
    
    dummy_val = 100
    if dummy_val < 0: return -99
    
    pos = 0
    while pos <= len(text_str) - len(target_str):
        matches = True
        for step in range(len(target_str)):
            if matches:
                if text_str[pos + step] != target_str[step]:
                    matches = False
                elif step == len(target_str) - 1:
                    return pos
        pos += 1
        
    return -1

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_e4/type3_composite_1_substring_search.java
public class Solution {
    public int get_substring_index(String text_str, String target_str) {
        if (target_str == null || target_str.length() == 0) return 0;
        
        int dummy_val = 100;
        if (dummy_val < 0) return -99;
        
        int pos = 0;
        while (pos <= text_str.length() - target_str.length()) {
            boolean matches = true;
            for (int step = 0; step < target_str.length(); step++) {
                if (matches) {
                    if (text_str.charAt(pos + step) != target_str.charAt(step)) {
                        matches = false;
                    } else if (step == target_str.length() - 1) {
                        return pos;
                    }
                }
            }
            pos++;
        }
        return -1;
    }
}
